brain-marks/CoreData/StorageProvider.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  StorageProvider.swift
+//  brain-marks
+//
+//  Created by Jay on 12/13/22.
+//
+
+import Foundation
+import CoreData
+import TelemetryClient
+
+class StorageProvider {
+
+    static let shared = StorageProvider()
+
+    let container: NSPersistentContainer
+
+    static var preview: StorageProvider = {
+        let controller = StorageProvider(inMemory: true)
+        for num in 0..<10 {
+            let category = CategoryEntity(context: controller.context)
+            category.name = "Category \(num)"
+            category.id = UUID()
+            category.dateCreated = Date()
+            category.dateModified = Date()
+            for tweetNum in 0..<5 {
+                let tweet = TweetEntity(context: controller.context)
+                tweet.id = UUID()
+                tweet.dateCreated = Date()
+                tweet.authorName = "heyjaywilson"
+                tweet.text = "this is a sample tweet \(tweetNum)"
+                category.addToTweets(tweet)
+            }
+        }
+        return controller
+    }()
+
+    init(inMemory: Bool = false) {
+        container = NSPersistentContainer(name: "BrainMarks")
+
+        if inMemory {
```

Comment:
what does this do?
Also considering this is a singleton, the `init` is supposed to be private, because you aren't supposed to have multiple instances of it at a time. Would this affect what the init does?

Reply:
I'll make the init private, but this tells the app which provider to look at for tests and such.

Reply:
Are there any unit tests for this? I didn't see any @heyjaywilson